org.eclipse.scanning.points/src/org/eclipse/scanning/jython/JythonInterpreterManager.java

```diff
@@ -0,0 +1,244 @@
+        state.path.add(new PyString(location.getAbsolutePath() + SCRIPTS));
+        
+        location = getBundleLocation("org.eclipse.scanning.sequencer");
+        state.path.add(new PyString(location.getAbsolutePath() + SCRIPTS));
+        
+        if (bundleNames!=null) {
+        	for (String bundle : bundleNames) {
+                location = getBundleLocation(bundle);
+                state.path.add(new PyString(location.getAbsolutePath() + SCRIPTS));
+			}
+        }
+	}
+
+
+	private static void initializePythonPath(ClassLoader loader) {
+		try {
+	    	String jythonBundleName = System.getProperty("org.eclipse.scanning.jython.osgi.bundle.name", "uk.ac.diamond.jython");
+	        File loc = getBundleLocation(jythonBundleName); // TODO Name the jython OSGi bundle without Diamond in it!
+	        if (loc ==null) return;
+	        File jythonDir = find(loc, "jython");
+	        if (jythonDir ==null) return;
+	        
+	        Properties props = new Properties();
+	    	props.setProperty("python.home", jythonDir.getAbsolutePath());
+	    	props.setProperty("python.console.encoding", "UTF-8"); // Used to prevent: console: Failed to install '': java.nio.charset.UnsupportedCharsetException: cp0.
+	    	props.setProperty("python.options.showJavaExceptions", "true");
+	    	props.setProperty("python.verbose", "warning");
+
+	    	Properties preprops = System.getProperties();
+	    	
+	    	PySystemState.initialize(preprops, props, new String[]{}, loader);
+	    
+	    } catch (Throwable ne) {
+	    	logger.debug("Problem loading jython bundles!", ne);
+	    }
+	}
+
+
+	private static ClassLoader createJythonClassLoader(ClassLoader classLoader) {
+		
+    	ClassLoader jythonClassloader = ScanPointGeneratorFactory.class.getClassLoader();
+    	
+    	try { // For non-unit tests, attempt to use the OSGi classloader of this bundle.
+    		String jythonBundleName = getJythonBundleName();
+    		CompositeClassLoader composite = new CompositeClassLoader(classLoader);
+   	  	    // Classloader for org.eclipse.scanning.points
+    		composite.addLast(ScanPointGeneratorFactory.class.getClassLoader());
+    		addLast(composite, jythonBundleName);
+    		jythonClassloader = composite;
+     		
+    	} catch (Throwable ne) {
+    		ne.printStackTrace();
+    		// Legal, if static classloader does not work in tests, there will be
+    		// errors. If bundle classloader does not work in product, there will be errors.
+    		// Typically the message is something like: 'cannot find module org.eclipse.scanning.api'
+    	}
+    	return jythonClassloader;
+	}
+
+	private static String getJythonBundleName() {
+		return System.getProperty("org.eclipse.scanning.jython.osgi.bundle.name", "uk.ac.diamond.jython");
+	}
+
+	private static void setJythonPaths(PySystemState state) {
+    	        	
+        try {
+        	// Search for the Libs directory which should have been expanded out either
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1135)